src/org/thoughtcrime/securesms/WelcomeActivity.java

```diff
@@ -167,7 +169,10 @@ private void startQrAccountCreation(String qrCode)
         DcHelper.getEventCenter(this).captureNextError();
 
         if (!dcContext.setConfigFromQr(qrCode)) {
-            progressError(DcHelper.getEventCenter(this).getCapturedError());
+            Util.sleep(100); // hack to avoid a race condition, see https://github.com/deltachat/deltachat-core-rust/issues/2787 for more details and possible fix
+            String err = DcHelper.getEventCenter(this).getCapturedError();
+            progressError(TextUtils.isEmpty(err) ? "Cannot create account from QR code." : err);
```

Comment:
untranslated text

Reply:
yip, but that text normally should not appear at all, if the function fails an error should be set.

(we have tons of untranslated error messages and such things)